### DISCUSSION TITLE
Optional parameter in where statement QueryBuilder

### DISCUSSION THREAD

**Discussion by bryan-vh** [mikro-orm/mikro-orm#2516] [Q&A]
If I have an optional parameter and I pass it to a where statement with QueryBuilder, will that where statement through an error or simply not be included in the SQL syntax? Or is the only way around that to do a check with an if else statement as seen below:

```
if (dto.offset) {
      qb.select(['*', likesCountQb, reservationsCountQb, friendsCountQb])
        .joinAndSelect('p.user', 'u')
        .where({ user: id })
        .andWhere('p.start > ?', [dto.offset])
        .orderBy({ start: 'ASC' })
        .limit(10)
} else {
      qb.select(['*', likesCountQb, reservationsCountQb, friendsCountQb])
        .joinAndSelect('p.user', 'u')
        .where({ user: id })
        .orderBy({ start: 'ASC' })
        .limit(10)
}
```

**Comment by bryan-vh** [comment]
Any update on this @B4nan?

**Reply by B4nan** [reply]
Not sure what you mean by optional statement, in context of QB there is no such thing, you either provide a condition or not. Especially if you want to use the weakly typed string variants like `where('...sql fragment...')`. If you try to do `where({ user: undefined })` it will be translated to `user is null`.

Note that QB is statefull, so you can simplify your example a lot, this should do the same work:

```ts
qb.select(['*', likesCountQb, reservationsCountQb, friendsCountQb])
        .joinAndSelect('p.user', 'u')
        .where({ user: id })
        .orderBy({ start: 'ASC' })
        .limit(10);

if (dto.offset) {
        qb.andWhere('p.start > ?', [dto.offset])
}
```
